### ISSUE TITLE
Querying by OCKOutcome.id not returning correct result

### ISSUE THREAD

**Issue by cbaker6** [carekit-apple/CareKit#430]
For some reason, I have unexpected results when using OCKOutcome.id. I can't seem to get back the exact outcome I'm looking for when using fetch which causes incorrect syncing. For example:

```swift
let careKitQuery = OCKOutcomeQuery(id: "x-coredata://842B1364-A6D8-42FE-BA15-D192D8D5EA45/OCKCDTask/p23_8")
                storeManager.store.fetchAnyOutcome(query: careKitQuery, callbackQueue: .global(qos: .background)){
                    result in
                    switch result{
                    case .success(let fetchedOutcome):
                        guard var mutableOutcome = fetchedOutcome as? OCKOutcome else{return}
                        mutableOutcome.remoteID = self.objectId
                        self.values.forEach{
                            for (index,value) in mutableOutcome.values.enumerated(){
                                guard let id = value.userInfo?[kPCKOutcomeValueUserInfoIDKey],
                                    id == $0.uuid else{
                                    continue
                                }
                                mutableOutcome.values[index].remoteID = $0.objectId
                            }
                        }
                        storeManager.store.updateAnyOutcome(mutableOutcome){
                            result in
                            switch result{
                            case .success(let updatedContact):
                                print("Updated remoteID of \(self.parseClassName): \(updatedContact)")
                            case .failure(let error):
                                print("Error updating remoteID. \(error)")
                            }
                        }
                    case .failure(let error):
                        print("Error in \(self.parseClassName).saveAndCheckRemoteID(). \(error)")
                    }
                }
```

This seems to return back random versions of OCKOutcome.id, for example on this one, I got back, `x-coredata://842B1364-A6D8-42FE-BA15-D192D8D5EA45/OCKCDTask/p23_7`.

**Comment by cbaker6** [comment]
A slightly related question about Outcome.uuid. It looks like the uuid changes for Outcome every time it's updated (liked the versioned entities, but I can see the reason for those). For Outcomes, shouldn't these stay the same? If they change, querying by uuid won't return the correct results when syncing to the Cloud.

My workaround is to keep a uuid that doesn't change in "tags" or "userInfo".

**Comment by erik-apple** [comment]
The uuid "changes" because updating an outcome actually results in the old outcome getting tombstoned, and a new outcome being created in its place. This is necessary to enable robust conflict detection and resolution.

I believe the right way to do what you want is to query an outcome by its `taskUUID` and `taskOccurrenceIndex`. These two in combination form an id that will have the behavior you're after (and that's actually what the `id` property is now). Unfortunately, I don't think we have a way to query against the `taskOccurrenceIndex` today -- that's probably something that merits adding.

**Comment by cbaker6** [comment]
Thanks @erik-apple! I didn't realize it was tombstoning until you mentioned it. I was able to adjust for this.

It seems the Outcome.id makes the correct `taskUUID` and `taskOccurenceIndex` string combination you mentioned. When syncing, I've adjusted to whenever I discover an `Outcome` not having a tag of `Outcome.id`, I tag it. In this way, I'm not able to query the initial Outcome created, but it's tombstoned anyways because of the updated tag. All subsequent outcomes can then be queried by their tag of `Outcome.id`
